### ISSUE TITLE
Range Picker Modal

### ISSUE THREAD

**Issue by FutureIronman7** [web-ridge/react-native-paper-dates#39]
when implementing range picker, the default color is dark purple. Is there a way to modify the color ?

**Comment by RichardLindhout** [comment]
You can edit the react native paper theme. https://callstack.github.io/react-native-paper/theming.html.

Maybe we'll add a custom style object or something in the future like react-native-paper-tabs

**Comment by FutureIronman7** [comment]
awesome, thanks a lot ! Will try that out

**Comment by IceDev-xyz** [comment]
If we are using react native paper theme, I think this component should accept the prop theme, as "Applying a theme to a paper component"

**Comment by RichardLindhout** [comment]
Yeah it will work the same as the Appbar component if I add it.

**Comment by ricohumme** [comment]
I would like this feature as well. Currently using this component in a non-paper project and my design doesn't match now.
Any thought on when this would be implemented?

**Comment by RichardLindhout** [comment]
> I would like this feature as well. Currently using this component in a non-paper project and my design doesn't match now.
> Any thought on when this would be implemented?

You can still wrap the react-native-paper theme in your App.js or around the DatePickerModal it'll work just fine + you can use other react-native-paper libraries then in the same styling :). But we'll soon merge this change and support the theme prop but I have some additional things I'd like to change with 'adaptive' and 'normal' mode.

That being said I'm really short in time a.t.m. so only critical fixes will get priority but I think I'll have more time in a few weeks!

**Comment by ricohumme** [comment]
Thanks, I couldn't get it to work first but now it does :)

**Comment by amireds** [comment]
> Thanks, I couldn't get it to work first but now it does :)

hi @RichardLindhout Can you be kind enough to explain how you implemented it pls. I added the theme feature from react paper, but I can't seem to figure out how to reference the element I want to change the color...

Thanks

**Comment by rrrhys** [comment]
> > Thanks, I couldn't get it to work first but now it does :)
> 
> hi @RichardLindhout Can you be kind enough to explain how you implemented it pls. I added the theme feature from react paper, but I can't seem to figure out how to reference the element I want to change the color...
> 
> Thanks

If you've currently got a
```
  <DatePickerModal
      mode="single"
      visible={visible}
      ... etc
    />
```

It needs to be

```

import { DefaultTheme, Provider as PaperProvider } from "react-native-paper";


const theme = { ...DefaultTheme, // add your stuff here }
<PaperProvider theme={theme}>
    <DatePickerModal
      mode="single"
      visible={visible} />
</PaperProvider>
```

**Comment by JohnBerd** [comment]
Can we re-open this please?
This component is the best for picking a range of dates. I don't use React Native Paper and I need to customize the colors.

**Comment by JB712** [comment]
I agree, I don't want to use react-native-paper, it will increase my app bundle for something that could be handled with a prop.
Please go for it @RichardLindhout  💚

**Comment by RichardLindhout** [comment]
You are definitely using react native paper if you use this library. It's a dependency of this project!
